### ISSUE TITLE
genome DB is unavailable

### ISSUE THREAD

**Issue by karl-cottenie** [ropensci/rentrez#196]
I tested these two links, and they worked as expected.	

https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=snp&id=6060535&retmote=rsr	
https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=pubmed&term=rentrez	

Here is my problem.

I received an error message for this search:
R> entrez_search(db = "genome", term = "sphingidae")
Error in ans[[1]] : subscript out of bounds

I checked if the genome database was listed:
R> entrez_dbs()
 [1] "pubmed"          "protein"         "nuccore"         "ipg"             "nucleotide"      "structure"      
 [7] "genome"          "annotinfo"       "assembly"        "bioproject"      "biosample"       "blastdbinfo"    
[13] "books"           "cdd"             "clinvar"         "gap"             "gapplus"         "grasp"          
[19] "dbvar"           "gene"            "gds"             "geoprofiles"     "medgen"          "mesh"           
[25] "nlmcatalog"      "omim"            "orgtrack"        "pmc"             "popset"          "proteinclusters"
[31] "pcassay"         "protfam"         "pccompound"      "pcsubstance"     "seqannot"        "snp"            
[37] "sra"             "taxonomy"        "biocollections"  "gtr"   

But then I found that the DB is unavailable:
R> entrez_db_summary(db = "genome")
 DbName: genome
 MenuName: Genome
 Description: Genomic sequences, contigs, and maps
 DbBuild: 
 Warning: pback220: DB is unavailable 

While the other databases are updated, e.g.:
R> entrez_db_summary(db = "taxonomy")
 DbName: taxonomy
 MenuName: Taxonomy
 Description: Taxonomy db
 DbBuild: Build240912-1410.1
 Count: 2744579
 LastUpdate: 2024/09/12 16:00 

I tested whether the genome database was available through the CLI, and that was the case:
$ datasets summary genome taxon 'bats' --assembly-source refseq --as-json-lines | dataformat tsv genome --fields accession,assminfo-name,annotinfo-name,annotinfo-release-date,organism-name
Assembly Accession      Assembly Name   Annotation Name Annotation Release Date Organism Name
GCF_004115265.2 mRhiFer1_v1.p   GCF_004115265.2-RS_2023_02      2023-02-27      Rhinolophus ferrumequinum
GCF_022682495.1 HLdesRot8A      GCF_022682495.1-RS_2023_02      2023-02-27      Desmodus rotundus
GCF_027574615.1 DD_ASM_mEF_20220401     GCF_027574615.1-RS_2023_03      2023-03-28      Eptesicus fuscus
GCF_004126475.2 mPhyDis1.pri.v3 NCBI Phyllostomus discolor Annotation Release 101       2020-08-31      Phyllostomus discolor
[...]

**In conclusion, it seems to me that the genome database is accessible, but not through the R entrez interface?**

**Comment by allenbaron** [comment]
Sorry. Can you clarify. Did you use esearch on the command line with the same inputs and it worked?

A quick debug shows that the error is returned from the entrez utils server. For me it returned: `Search Backend failed: Database is not supported: genome`.

`entrez_db_searchable("genome")` also returns nothing. And `entrez_info("genome") returns only links.

**Comment by karl-cottenie** [comment]
I accessed the genome database through the CLI following these instructions: https://www.ncbi.nlm.nih.gov/datasets/docs/v2/how-tos/genomes/get-genome-metadata/

with the following command

datasets summary genome taxon 'bats' [... the rest of the command is just to reformat the output ...]

That command did give me output

Assembly Accession Assembly Name Annotation Name Annotation Release Date Organism Name
GCF_004115265.2 mRhiFer1_v1.p GCF_004115265.2-RS_2023_02 2023-02-27 Rhinolophus ferrumequinum
GCF_022682495.1 HLdesRot8A GCF_022682495.1-RS_2023_02 2023-02-27 Desmodus rotundus
GCF_027574615.1 DD_ASM_mEF_20220401 GCF_027574615.1-RS_2023_03 2023-03-28 Eptesicus fuscus
GCF_004126475.2 mPhyDis1.pri.v3 NCBI Phyllostomus discolor Annotation Release 101 2020-08-31 Phyllostomus discolor
[...]

So I just assumed that the genome database is accessible? Unless the database called "genome" through the CLI "dataset" command is different from the database called "genome" accessed through R?

**Comment by allenbaron** [comment]
That looks like a different API. `rentrez` uses NCBI's Entrez Utilities (https://www.ncbi.nlm.nih.gov/home/tools/).

**Comment by karl-cottenie** [comment]
I did not know that, thanks for the clarification. Thus the problem that the function entrez_db_summary(db = "genome") returns nothing is on the NCBI end, and there is nothing you can do about that?

**Comment by allenbaron** [comment]
That is correct. The issues you are having are all directly because of the data provided by NCBI's servers and not `rentrez`.

NCBI has replaced the webpages corresponding to these databases with the datasets API you've mentioned. I can't tell you for sure if access to this database via `rentrez`/E-utilities has changed or not, since I don't use it regularly, but it is supposed to have stayed the same according to their documentation: https://support.nlm.nih.gov/kbArticle/?pn=KA-05455.

**Comment by theo-allnutt-bioinformatics** [comment]
For me the genome db is still not working through esearch but there seems to be almost no mention of it online.
e.g.
esearch -db genome -query "Bacteria"
<ENTREZ_DIRECT>
  <Db>genome</Db>
  <WebEnv>MCID_68e5ead9f5229810e002a458</WebEnv>
  <QueryKey>1</QueryKey>
  <Count>0</Count>
  <Step>1</Step>
</ENTREZ_DIRECT>

No records returned.
At the moment, NCBI will not answer any queries.

**Comment by allenbaron** [comment]
_NCBI probably hasn't responded because no one is at work. I imagine they're all out given the current government shutdown._

I think it's unlikely that NCBI will do anything to make the genome database work with eutils/rentrez. My impression is that they see the Entrez Utils as an old system headed toward obsoletion and replacement by the newer Datasets API (based on a conversion I had with someone from NCBI recently).

As far as I'm aware, there isn't currently an R package to support access to the Datasets API. It's possible to make one automatically with [openapi-generator](https://openapi-generator.tech/) as described in the [NCBI Datasets API documentation](https://www.ncbi.nlm.nih.gov/datasets/docs/v2/api/languages/), but in my limited experience the code produced is less than ideal and may not work.
